onnx/common/ir_pb_converter.cc

```diff
@@ -530,6 +537,58 @@ void encodeGraph(ONNX_NAMESPACE::GraphProto * p_g, const std::shared_ptr<Graph>
 void ExportModelProto(ONNX_NAMESPACE::ModelProto* p_m, const std::shared_ptr<Graph>& g) {
   ONNX_NAMESPACE::GraphProto* p_g = p_m->mutable_graph();
   encodeGraph(p_g, g);
+  // Add new opset_versions
+  p_m->clear_opset_import();
+  for (OpSetID opset : g->opset_versions) {
+    OperatorSetIdProto *opset_version_output = p_m->add_opset_import();
+    opset_version_output->set_domain(opset.domain);
+    opset_version_output->set_version(opset.version);
+  }
+}
+
+ONNX_NAMESPACE::ModelProto PrepareOutput(const ONNX_NAMESPACE::ModelProto& mp_in) {
+  ONNX_NAMESPACE::ModelProto mp_out{};
+
+  if (mp_in.has_ir_version()) {
+    mp_out.set_ir_version(mp_in.ir_version());
+  }
+  if (mp_in.has_producer_name()) {
+    mp_out.set_producer_name(mp_in.producer_name());
+  }
+  if (mp_in.has_producer_version()) {
```

Comment:
Yeah, I think that makes sense!  @houseroad who would be the best person from the Optimizer project to verify this with?

Reply:
Fwiw, we can entirely bypass using PrepareOutput in the VersionConverter and instead just copy mp_in directly in convert_version

Reply:
Ok so here's the design issue right now: If we attempt to refill the properties based on the new graph, we face the problem of re-adding existing inputs/outputs/attributes.  As such, to appropriately implement this, we would need to write a new version of ExportModelProto which replaces elements/nodes rather than adding them.  If @houseroad thinks this could be useful, I'd be happy to do so.

Reply:
This is a legacy problem. Let's fix it in a different pr.